sdk/nodejs/resource.ts

```diff
@@ -454,8 +455,16 @@ export abstract class Resource {
             }
         }
 
+        // Store the provider in `__prov` if the provider's package is the same
+        // as the resource's package. Otherwise, don't, as the user specified
+        // `provider: someProvider` as shorthand for `providers: [someProvider]`.
```

Comment:
Can you make this comment be more high-level explaining the motivation behind this in comment form for future readers?

Reply:
I've explained more in the comment.